File: src/test/java/removeelement/SolutionTest.java
package removeelement;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SolutionTest {
    private final Solution solution = new Solution();

    @Test
    public void testOne() {
        int[] nums = {3, 2, 2, 3};
        int val = 3;

        int remainingElements = solution.removeElement(nums, val);

        assertThat(remainingElements).isEqualTo(2);
    }

    @Test
    public void testTwo() {
        int[] nums = {0, 1, 2, 2, 3, 0, 4, 2};
        int val = 2;

        int remainingElements = solution.removeElement(nums, val);


        assertThat(remainingElements).isEqualTo(5);
    }

    @Test
    public void shouldPassWhenNoMatchingElements() {
        int[] nums = {1, 2, 3, 4, 5, 6, 7};
        int val = 8;

        int remainingElements = solution.removeElement(nums, val);

        assertThat(remainingElements).isEqualTo(nums.length);
    }
}

File: src/test/kotlin/dev/sanket/mergetwosortedlists/SolutionTest.kt
package dev.sanket.mergetwosortedlists

import assertk.assertThat
import assertk.assertions.isEqualTo
import assertk.assertions.isNull
import dev.sanket.dev.sanket.common.ListNode
import dev.sanket.dev.sanket.mergetwosortedlists.Solution
import org.junit.jupiter.api.Test

class SolutionTest {

    private val solution = Solution()

    @Test
    fun shouldReturnNullWhenBothListsAreNull() {
        val mergedList = solution.mergeTwoLists(null, null)

        assertThat(mergedList).isNull()
    }

    @Test
    fun shouldReturnListOneWhenListTwoIsNull() {
        val listOne = ListNode(1).next(ListNode(3).next(ListNode(4)))

        val mergedList = solution.mergeTwoLists(listOne, null)

        var currentNode = mergedList
        arrayOf(1, 3, 4).forEach {
            assertThat(currentNode?.`val`).isEqualTo(it)
            currentNode = currentNode?.next
        }
    }

    @Test
    fun shouldReturnListTwoWhenListOneIsNull() {
        val listTwo = ListNode(1).next(ListNode(3).next(ListNode(4)))

        val mergedList = solution.mergeTwoLists(null, listTwo)

        var currentNode = mergedList
        arrayOf(1, 3, 4).forEach {
            assertThat(currentNode?.`val`).isEqualTo(it)
            currentNode = currentNode?.next
        }
    }

    @Test
    fun testOne() {
        val listOne = ListNode(1).next(ListNode(2).next(ListNode(4)))
        val listTwo = ListNode(1).next(ListNode(3).next(ListNode(4)))

        val mergedList = solution.mergeTwoLists(listOne, listTwo)

        var currentNode = mergedList
        arrayOf(1, 1, 2, 3, 4, 4).forEach {
            assertThat(currentNode?.`val`).isEqualTo(it)
            currentNode = currentNode?.next
        }
    }
}